docs/content/workflows/pipelines/actions/plugins/write-plugin.md

```diff
@@ -10,10 +10,16 @@ A step is :
 
 * a builtin action, as GitClone, etc... [read more]({{< relref "workflows/pipelines/actions/builtin/_index.md" >}})
 * a user action [read more]({{< relref "workflows/pipelines/actions/user-actions/_index.md" >}})
-* a Plugin Action
+* a plugin action
 
-A Plugin is a Golang Binary.
+A Plugin is simply an executable which expose a GRPC server corresponding to the right [proto file](https://github.com/ovh/cds/tree/master/sdk/grpcplugin/actionplugin/actionplugin.proto). You can use any languages. The CDS worker will simply query in GRPC the plugin (which is the GRPC server).
 
-Take a look at https://github.com/ovh/cds/tree/master/sdk/plugin/dummy/dummy_plugin.go
+So a CDS plugin have these requirements in order to communicate with worker:
 
-Contribute on https://github.com/ovh/cds/tree/master/contrib/plugins
++ Must expose a GRPC server
++ Must implement methods and messages coming from this [proto file](https://github.com/ovh/cds/tree/master/sdk/grpcplugin/actionplugin/actionplugin.proto)
++ And last but not least at the launch of your plugin you can use random unix socket or random tcp port but in order to inform worker what is your address you have to display this log at the launch of your plugin `XXX is ready to accept new connection` where `XXX` is your ip address with port or your unix socket (example: `127.0.0.1:55939 is ready to accept new connection`).
```

Comment:
- Display this log at the launch of your plugin `XXX is ready to accept new connection` where `XXX` is your ip address with port or your unix socket (example: `127.0.0.1:55939 is ready to accept new connection`). Note that your plugin can use any unix socket or tcp port as long as it informs the worker using the log line above.

Reply:
[technical] I find this way of interaction a bit ugly :)

Reply:
Yes agree with you, it's not the best way. I'll work for a better way later. At the moment I try to stay ISO as before